core/src/Xios.cpp

```diff
@@ -213,69 +275,517 @@ void Xios::setCalendarTimestep(cxios_duration timestep)
+    std::string fieldName(cStr, size);
+    boost::algorithm::trim_right(fieldName);
+    return fieldName;
+}
+
+/*!
+ * Get the operation associated with a field with a given ID
+ *
+ * @param the field ID
+ * @return operation used for the corresponding field
+ */
+std::string Xios::getFieldOperation(std::string fieldId)
+{
+    int size = 20;
+    char cStr[size];
+    xios::CField* field = getField(fieldId);
+    cxios_get_field_operation(field, cStr, size);
+    std::string operation(cStr, size);
+    boost::algorithm::trim_right(operation);
+    return operation;
+}
+
+/*!
+ * Get the grid reference associated with a field with a given ID
+ *
+ * @param the field ID
+ * @return grid reference used for the corresponding field
+ */
+std::string Xios::getFieldGridRef(std::string fieldId)
+{
+    int size = 20;
+    char cStr[size];
+    xios::CField* field = getField(fieldId);
+    cxios_get_field_grid_ref(field, cStr, size);
+    std::string gridRef(cStr, size);
+    boost::algorithm::trim_right(gridRef);
+    return gridRef;
+}
+
+/*!
+ * Verify whether a name has been defined for a given field ID
+ *
+ * @param the field ID
+ * @return `true` if the name has been set, otherwise `false`
+ */
+bool Xios::isDefinedFieldName(std::string fieldId)
+{
+    xios::CField* field = getField(fieldId);
+    return cxios_is_defined_field_name(field);
+}
+
+/*!
+ * Verify whether an operation has been defined for a given field ID
+ *
+ * @param the field ID
+ * @return `true` if the operation has been set, otherwise `false`
+ */
+bool Xios::isDefinedFieldOperation(std::string fieldId)
+{
+    xios::CField* field = getField(fieldId);
+    return cxios_is_defined_field_operation(field);
+}
+
+/*!
+ * Verify whether a grid reference has been defined for a given field ID
+ *
+ * @param the field ID
+ * @return `true` if the grid reference has been set, otherwise `false`
+ */
+bool Xios::isDefinedFieldGridRef(std::string fieldId)
+{
+    xios::CField* field = getField(fieldId);
+    return cxios_is_defined_field_grid_ref(field);
+}
+
+/*!
+ * Get the grid associated with a given ID
+ *
+ * @param the grid ID
+ * @return a pointer to the XIOS CGrid object
+ */
+xios::CGrid* Xios::getGrid(std::string gridId)
+{
+    xios::CGrid* grid = NULL;
+    cxios_grid_handle_create(&grid, gridId.c_str(), gridId.length());
+    return grid;
+}
+
+/*!
+ * Get the name of a grid with a given ID
+ *
+ * @param the grid ID
+ * @return name of the corresponding grid
+ */
+std::string Xios::getGridName(std::string gridId)
+{
+    int size = 20;
+    char cStr[size];
+    xios::CGrid* grid = getGrid(gridId);
+    cxios_get_grid_name(grid, cStr, size);
+    std::string gridName(cStr, size);
+    boost::algorithm::trim_right(gridName);
+    return gridName;
+}
+
+/*!
+ * Get the file associated with a given ID
+ *
+ * @param the file ID
+ * @return a pointer to the XIOS CFile object
+ */
+xios::CFile* Xios::getFile(std::string fileId)
+{
+    xios::CFile* file = NULL;
```

Comment:
just a thought, not necessarily for this PR. Would it be sensible to throw an exception if any of the getter objects are still `NULL`? Ill be honest... I didn't really check what happens if `xios` getters fail. If `xios` throws the error anyway then it may not be necessary.

Reply:
Good point. I've opened #562 to address this.